# Accquire

**imports**

In [2]:
import pandas as pd

# import sys to use system functions
# this allow for imports form directories different from the current
import sys
sys.path.append("../util_")
# import acquire_

## Get data

Data from [catalog.data.gov](https://catalog.data.gov/dataset/motor-vehicle-collisions-crashes)

In [10]:
# get data
# low memory set to false beacuse some columns have unsoecified datatypes
vehicle = pd.read_csv("project_data/Motor_Vehicle_Collisions_-_Crashes.csv", low_memory=False)

# look at my data
vehicle.head().T

0  \
CRASH DATE                                       09/11/2021   
CRASH TIME                                             2:39   
BOROUGH                                                 NaN   
ZIP CODE                                                NaN   
LATITUDE                                                NaN   
LONGITUDE                                               NaN   
LOCATION                                                NaN   
ON STREET NAME                        WHITESTONE EXPRESSWAY   
CROSS STREET NAME                                 20 AVENUE   
OFF STREET NAME                                         NaN   
NUMBER OF PERSONS INJURED                               2.0   
NUMBER OF PERSONS KILLED                                0.0   
NUMBER OF PEDESTRIANS INJURED                             0   
NUMBER OF PEDESTRIANS KILLED                              0   
NUMBER OF CYCLIST INJURED                                 0   
NUMBER OF CYCLIST KILLED                                  0   
NUMBER OF MOTORIST INJURED                                2   
NUMBER OF MOTORIST KILLED                                 0   
CONTRIBUTING FACTOR VEHICLE 1  Aggressive Driving/Road Rage   
CONTRIBUTING FACTOR VEHICLE 2                   Unspecified   
CONTRIBUTING FACTOR VEHICLE 3                           NaN   
CONTRIBUTING FACTOR VEHICLE 4                           NaN   
CONTRIBUTING FACTOR VEHICLE 5                           NaN   
COLLISION_ID                                        4455765   
VEHICLE TYPE CODE 1                                   Sedan   
VEHICLE TYPE CODE 2                                   Sedan   
VEHICLE TYPE CODE 3                                     NaN   
VEHICLE TYPE CODE 4                                     NaN   
VEHICLE TYPE CODE 5                                     NaN   

                                                     1                      2  \
CRASH DATE                                  03/26/2022             06/29/2022   
CRASH TIME                                       11:45                   6:55   
BOROUGH                                            NaN                    NaN   
ZIP CODE                                           NaN                    NaN   
LATITUDE                                           NaN                    NaN   
LONGITUDE                                          NaN                    NaN   
LOCATION                                           NaN                    NaN   
ON STREET NAME                 QUEENSBORO BRIDGE UPPER     THROGS NECK BRIDGE   
CROSS STREET NAME                                  NaN                    NaN   
OFF STREET NAME                                    NaN                    NaN   
NUMBER OF PERSONS INJURED                          1.0                    0.0   
NUMBER OF PERSONS KILLED                           0.0                    0.0   
NUMBER OF PEDESTRIANS INJURED                        0                      0   
NUMBER OF PEDESTRIANS KILLED                         0                      0   
NUMBER OF CYCLIST INJURED                            0                      0   
NUMBER OF CYCLIST KILLED                             0                      0   
NUMBER OF MOTORIST INJURED                           1                      0   
NUMBER OF MOTORIST KILLED                            0                      0   
CONTRIBUTING FACTOR VEHICLE 1        Pavement Slippery  Following Too Closely   
CONTRIBUTING FACTOR VEHICLE 2                      NaN            Unspecified   
CONTRIBUTING FACTOR VEHICLE 3                      NaN                    NaN   
CONTRIBUTING FACTOR VEHICLE 4                      NaN                    NaN   
CONTRIBUTING FACTOR VEHICLE 5                      NaN                    NaN   
COLLISION_ID                                   4513547                4541903   
VEHICLE TYPE CODE 1                              Sedan                  Sedan   
VEHICLE TYPE CODE 2                                NaN       

## Understand the data

In [7]:
vehicle.shape # data size

(2005011, 29)

In [8]:
vehicle.columns # visual look of the data types

Index(['CRASH DATE', 'CRASH TIME', 'BOROUGH', 'ZIP CODE', 'LATITUDE',
       'LONGITUDE', 'LOCATION', 'ON STREET NAME', 'CROSS STREET NAME',
       'OFF STREET NAME', 'NUMBER OF PERSONS INJURED',
       'NUMBER OF PERSONS KILLED', 'NUMBER OF PEDESTRIANS INJURED',
       'NUMBER OF PEDESTRIANS KILLED', 'NUMBER OF CYCLIST INJURED',
       'NUMBER OF CYCLIST KILLED', 'NUMBER OF MOTORIST INJURED',
       'NUMBER OF MOTORIST KILLED', 'CONTRIBUTING FACTOR VEHICLE 1',
       'CONTRIBUTING FACTOR VEHICLE 2', 'CONTRIBUTING FACTOR VEHICLE 3',
       'CONTRIBUTING FACTOR VEHICLE 4', 'CONTRIBUTING FACTOR VEHICLE 5',
       'COLLISION_ID', 'VEHICLE TYPE CODE 1', 'VEHICLE TYPE CODE 2',
       'VEHICLE TYPE CODE 3', 'VEHICLE TYPE CODE 4', 'VEHICLE TYPE CODE 5'],
      dtype='object')

In [12]:
vehicle.info() # dataframe imformation

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2005011 entries, 0 to 2005010
Data columns (total 29 columns):
 #   Column                         Dtype  
---  ------                         -----  
 0   CRASH DATE                     object 
 1   CRASH TIME                     object 
 2   BOROUGH                        object 
 3   ZIP CODE                       object 
 4   LATITUDE                       float64
 5   LONGITUDE                      float64
 6   LOCATION                       object 
 7   ON STREET NAME                 object 
 8   CROSS STREET NAME              object 
 9   OFF STREET NAME                object 
 10  NUMBER OF PERSONS INJURED      float64
 11  NUMBER OF PERSONS KILLED       float64
 12  NUMBER OF PEDESTRIANS INJURED  int64  
 13  NUMBER OF PEDESTRIANS KILLED   int64  
 14  NUMBER OF CYCLIST INJURED      int64  
 15  NUMBER OF CYCLIST KILLED       int64  
 16  NUMBER OF MOTORIST INJURED     int64  
 17  NUMBER OF MOTORIST KILLED      int64  
 18  CO

In [13]:
vehicle.describe() # Descriptive statistics

LATITUDE     LONGITUDE  NUMBER OF PERSONS INJURED  \
count  1.776288e+06  1.776288e+06               2.004993e+06   
mean   4.062781e+01 -7.375243e+01               3.004145e-01   
std    1.980445e+00  3.727679e+00               6.914412e-01   
min    0.000000e+00 -2.013600e+02               0.000000e+00   
25%    4.066795e+01 -7.397494e+01               0.000000e+00   
50%    4.072106e+01 -7.392732e+01               0.000000e+00   
75%    4.076948e+01 -7.386665e+01               0.000000e+00   
max    4.334444e+01  0.000000e+00               4.300000e+01   

       NUMBER OF PERSONS KILLED  NUMBER OF PEDESTRIANS INJURED  \
count              2.004980e+06                   2.005011e+06   
mean               1.436423e-03                   5.501915e-02   
std                3.994945e-02                   2.409417e-01   
min                0.000000e+00                   0.000000e+00   
25%                0.000000e+00                   0.000000e+00   
50%                0.000000e+00                   0.000000e+00   
75%                0.000000e+00                   0.000000e+00   
max                8.000000e+00                   2.700000e+01   

       NUMBER OF PEDESTRIANS KILLED  NUMBER OF CYCLIST INJURED  \
count                  2.005011e+06               2.005011e+06   
mean                   7.246843e-04               2.586918e-02   
std                    2.740606e-02               1.606470e-01   
min                    0.000000e+00               0.000000e+00   
25%                    0.000000e+00               0.000000e+00   
50%                    0.000000e+00               0.000000e+00   
75%                    0.000000e+00               0.000000e+00   
max                    6.000000e+00               4.000000e+00   

       NUMBER OF CYCLIST KILLED  NUMBER OF MOTORIST INJURED  \
count              2.005011e+06                2.005011e+06   
mean               1.112213e-04                2.165948e-01   
std                1.059276e-02                6.526298e-01   
min                0.000000e+00                0.000000e+00   
25%                0.000000e+00                0.000000e+00   
50%                0.000000e+00                0.000000e+00   
75%                0.000000e+00                0.000000e+00   
max                2.000000e+00                4.300000e+01   

       NUMBER OF MOTORIST KILLED  COLLISION_ID  
count               2.005011e+06  2.005011e+06  
mean                5.820417e-04  3.106344e+06  
std                 2.633293e-02  1.503778e+06  
min                 0.000000e+00  2.200000e+01  
25%                 0.000000e+00  3.137372e+06  
50%                 0.000000e+00  3.638723e+06  
75%                 0.000000e+00  4.140228e+06  
max                 5.000000e+00  4.641780e+06

In [14]:
# count of numeric columns and object columns
numerics = len(vehicle.select_dtypes("number").columns)
objects = len(vehicle.select_dtypes("object").columns)

print("Numeric col count:", numerics)
print("object col count:", objects)

Numeric col count: 11
object col count: 18


**What I see:**

- I have 2005011 rows and 29 columns
- 11 of the 29 columns are numric while 18 of them are string object colums
- I have unsure null values in my data
- I also see the descriptive statistics of my data
- Below is the data dictionary which helps me understand the features of my data

## Data dictionary
|Column Name|Description|Data Type 
|----|----|----|
|**CRASH DATE**|Occurrence date of collision|Floating Timestamp|
|**CRASH TIME**|Occurrence time of collision|Plain Text|
|**BOROUGH**|Borough where collision occurred|Plain Text|
|**ZIP CODE**|Postal code of incident occurrence|Plain Text|
|**LATITUBE**|Latitude coordinate for Global Coordinate System, WGS 1984, decimal degrees (EPSG 4326)|Number|
|**LONGITUBE**|Longitude coordinate for Global Coordinate System, WGS 1984, decimal degrees (EPSG 4326)|Number|
|**LOCATION**|Latitude , Longitude pair|Location|
|**ON STREET NAME**|Street on which the collision occurred|Plain Text|
|**CROSS STREET NAME**|Nearest cross street to the collision|Plain Text|
|**OFF STREET NAME**|Street address if known|Plain Text|
|**NUMBER OF PERSONS INJURED**|Number of persons injured|Number|
|**NUMBER OF PERSONS KILLED**|Number of persons killed|Number|
|**NUMBER OF PEDESTRIANS INJURED**| Number of pedestrians injured|Number|
|**NUMBER OF PEDESTRIANS KILLED**|Number of pedestrians killed|Number|
|**NUMBER OF CYCLIST INJURED**|Number of cyclists injured|Number|
|**NUMBER OF CYCLIST KILLED**|Number of cyclists killed|Number|
|**NUMBER OF MOTORIST INJURED**|Number of vehicle occupants injured|Number|
|**NUMBER OF MOTORIST KILLED**|Number of vehicle occupants killed|Number|
|**CONTRIBUTING FACTOR VEHICLE 1**|Factors contributing to the collision for Vehicle 1|Plain Text|
|**CONTRIBUTING FACTOR VEHICLE 2**|Factors contributing to the collision for Vehicle 2|Plain Text|
|**CONTRIBUTING FACTOR VEHICLE 3**|Factors contributing to the collision for Vehicle 3|Plain Text|
|**CONTRIBUTING FACTOR VEHICLE 4**|Factors contributing to the collision for Vehicle 4|Plain Text|
|**CONTRIBUTING FACTOR VEHICLE 5**|Factors contributing to the collision for Vehicle 5|Plain Tex|
|**COLLISION_ID**|Unique record code generated by system. Primary Key for Crash table.|Nuumber|
|**VEHICLE TYPE CODE 1**|Type of vehicle based on the selected vehicle category (ATV, bicycle, car/suv, ebike, escooter, truck/bus, motorcycle, other)|Plain Tex|
|**VEHICLE TYPE CODE 2**|Type of vehicle based on the selected vehicle category (ATV, bicycle, car/suv, ebike, escooter, truck/bus, motorcycle, other)|Plain Tex|
|**VEHICLE TYPE CODE 3**|Type of vehicle based on the selected vehicle category (ATV, bicycle, car/suv, ebike, escooter, truck/bus, motorcycle, other)|Plain Tex|
|**VEHICLE TYPE CODE 4**|Type of vehicle based on the selected vehicle category (ATV, bicycle, car/suv, ebike, escooter, truck/bus, motorcycle, other)|Plain Tex|
|**VEHICLE TYPE CODE 5**|Type of vehicle based on the selected vehicle category (ATV, bicycle, car/suv, ebike, escooter, truck/bus, motorcycle, other)|Plain Tex|
